be/src/storage/lake/schema_change.cpp

```diff
@@ -0,0 +1,351 @@
+
+    DISALLOW_COPY_AND_MOVE(LinkedSchemaChange);
+
+    Status init() override { return Status::OK(); }
+    Status process(RowsetPtr rowset, RowsetMetadata* new_rowset_metadata) override;
+};
+
+class ConvertedSchemaChange : public SchemaChange {
+public:
+    explicit ConvertedSchemaChange(Tablet* base_tablet, Tablet* new_tablet, int64_t version,
+                                   ChunkChanger* chunk_changer)
+            : SchemaChange(),
+              _base_tablet(base_tablet),
+              _new_tablet(new_tablet),
+              _version(version),
+              _chunk_changer(chunk_changer) {}
+    ~ConvertedSchemaChange() override = default;
+
+    Status init() override;
+
+protected:
+    Tablet* _base_tablet = nullptr;
+    Tablet* _new_tablet = nullptr;
+    int64_t _version = 0;
+    ChunkChanger* _chunk_changer = nullptr;
+
+    TabletReaderParams _read_params;
+    std::shared_ptr<const TabletSchema> _new_tablet_schema;
+    Schema _base_schema;
+    Schema _new_schema;
+    ChunkPtr _base_chunk;
+    ChunkPtr _new_chunk;
+    std::vector<size_t> _char_field_indexes;
+    std::unique_ptr<MemPool> _mem_pool;
+};
+
+class DirectSchemaChange final : public ConvertedSchemaChange {
+public:
+    explicit DirectSchemaChange(Tablet* base_tablet, Tablet* new_tablet, int64_t version, ChunkChanger* chunk_changer)
+            : ConvertedSchemaChange(base_tablet, new_tablet, version, chunk_changer) {}
+    ~DirectSchemaChange() override = default;
+
+    DISALLOW_COPY_AND_MOVE(DirectSchemaChange);
+
+    Status process(RowsetPtr rowset, RowsetMetadata* new_rowset_metadata) override;
+};
+
+class SortedSchemaChange final : public ConvertedSchemaChange {
+public:
+    explicit SortedSchemaChange(Tablet* base_tablet, Tablet* new_tablet, int64_t version, ChunkChanger* chunk_changer,
+                                size_t memory_limitation)
+            : ConvertedSchemaChange(base_tablet, new_tablet, version, chunk_changer),
+              _memory_limitation(memory_limitation) {}
+    ~SortedSchemaChange() override = default;
+
+    DISALLOW_COPY_AND_MOVE(SortedSchemaChange);
+
+    Status init() override;
+    Status process(RowsetPtr rowset, RowsetMetadata* new_rowset_metadata) override;
+
+private:
+    size_t _memory_limitation;
+    size_t _max_buffer_size;
+    std::unique_ptr<std::vector<uint32_t>> _selective;
+};
+
+Status LinkedSchemaChange::process(RowsetPtr rowset, RowsetMetadata* new_rowset_metadata) {
+    new_rowset_metadata->CopyFrom(rowset->metadata());
+    return Status::OK();
+}
+
+Status ConvertedSchemaChange::init() {
+    _read_params.reader_type = ReaderType::READER_ALTER_TABLE;
+    _read_params.skip_aggregation = false;
+    _read_params.chunk_size = config::vector_chunk_size;
+    _read_params.use_page_cache = false;
+
+    ASSIGN_OR_RETURN(auto base_tablet_schema, _base_tablet->get_schema());
+    _base_schema = ChunkHelper::convert_schema_to_format_v2(*base_tablet_schema,
```

Comment:
In the lake, format v1 is also not supported. It's not necessary to convert the format.

Reply:
Remove this later.